src/x-terminal.js

```diff
@@ -510,6 +511,20 @@ class XTerminalSingleton {
 	unfocus () {
 		atom.views.getView(atom.workspace).focus()
 	}
+	
+	focus () {
+        // if non exist, then open one
+		if (this.terminals_set.size == 0) {
+            this.open(
+                this.profilesSingleton.generateNewUri(),
+                this.addDefaultPosition(),
+            )
+        // otherwise find the most recently active terminal and focus on it
+        } else {
+            const sortedTerminals = ([...this.terminals_set]).sort((a,b)=>a.lastActivity - b.lastActivity)
+            sortedTerminals[0].focusOnTerminal()
```

Comment:
terminals have a property `activeIndex` that might be helpful here instead of adding another property.

```suggestion
            const activeTerminal = ([...this.terminals_set]).find(t => t.activeIndex === 0)
            activeTerminal.focusOnTerminal()
```